backend/src/api/routers/chains.py

```diff
@@ -21,11 +21,18 @@ class UserInput(BaseModel):
 
 
 load_dotenv()
+
 use_cuda = os.getenv("USE_CUDA")
 
-llm = ChatGoogleGenerativeAI(model="gemini-pro", temperature=1)
-hf_embdeddings = "BAAI/bge-large-en-v1.5"
-hf_reranker = "BAAI/bge-reranker-base"
+if os.getenv("GOOGLE_GEMINI") == "1_pro":
+    llm = ChatGoogleGenerativeAI(model="gemini-pro", temperature=1)
+elif os.getenv("GOOGLE_GEMINI") == "1.5_flash":
+    llm = ChatVertexAI(model_name="gemini-1.5-flash")
+elif os.getenv("GOOGLE_GEMINI") == "1.5_pro":
+    llm = ChatVertexAI(model_name="gemini-1.5-pro")
+
+hf_embdeddings = os.getenv("HF_EMBEDDINGS")
+hf_reranker = os.getenv("HF_RERANKER")
```

Comment:
Should it error out if environment variables are not provided. Or does not match the correct model?

Reply:
I've added a check for the env vars